lib/server.js

```diff
@@ -386,6 +406,88 @@ module.exports = function createServer(config, log) {
     }
   )
 
+  api.post('/checkCustomRule', function(req, res, next) {
+    const action = req.body.action
+    const ip = req.body.ip
+    const email = req.body.email
+
+    if (!ip || !action || !email) {
+      const err = {code: 'MissingParameters', message: 'action, ip email, are all required'}
+      log.error({op: 'request.checkCustomRule', action, ip, email})
+      res.send(400, err)
+      return next()
+    }
+
+    // Get all the custom rules that might apply to this action
+    const rules = config.userDefinedRateLimitRules
+    const checkRules = computedRules.get(action)
+
+    if (checkRules.length <= 0) {
+      const err = {code: 'MissingParameters', message: 'no rules found for ' + action}
+      log.error({op: 'request.checkCustomRule', action, ip, email})
+      res.send(400, err)
+      return next()
+    }
+
+    // Retrieve the record for each rule and check if `action` has been rate limited
+    const recordResults = []
+    return P.each(checkRules, (ruleName) => {
+      let retryAfter = null
+      const recordKey = ip + utils.createHashHex(email, ruleName)
```

Comment:
Is there any value in keeping IP as part of memcache key? For example, would we ever need to delete user defined rules for a specific IP?

Reply:
I can't think of a reason why we'd want to do this off the top of my head.  I also don't recall there being a way to delete keys from memcached by prefix, so in order to delete the key, you'd have to know all the inputs to the key calculation anyway.

I think we're more likely to want to have the ruleName in cleartext than either the IP or email, which would make this something like `recordKey = ruleName + ":" + utils.createHashHex(ip, email)` or similar.